### ISSUE TITLE
Getting warning about prop type for focused prop passed to DayPickerSingleDateController component

### ISSUE THREAD

**Issue by meerim1987** [airbnb/react-dates#2136]
**react-dates version**
react-dates@21.8.0

**Describe the bug**
Getting warning about prop type for focused prop passed toDayPickerSingleDateController component. 
```Warning: Failed prop type: Invalid prop `focused` of type `object` supplied to `DayPickerSingleDateController`, expected `boolean`.```
Tried to pass boolean, it stops complaining but then the component focus stops working and component behaves weirdly by losing its focus. 

**Source code (including props configuration)**
```
const [focusedDate, setFocused] = useState(true);

 <DayPickerSingleDateController
          date={selectedDate}
          onDateChange={dateValue => setSelectedDate(dateValue)}
          hideKeyboardShortcutsPanel
          onOutsideClick={onOutsideClick}
          onClose={timePickerOnClose}
          isOutsideRange={day => day.isBefore(firstValidDate, "day")}
          focused={focusedDate}
          onFocusChange={focused => {
            setFocused(focused);
          }}
        />
```
Like in the above code the component focus works, but warning appears. 
Tried to set it to boolean `onFocusChange={({ focused }) => {
            setFocused(focused);
          }}` .  In that case it stops complaining, but then focus stops working for this component. 

**Screenshots/Gifs**
![Screen Shot 2021-07-23 at 10 58 25 AM](https://user-images.githubusercontent.com/44931882/126835732-1a8d6dba-d829-49e4-8271-2dabc8698cd4.png)

**Comment by ljharb** [comment]
When `setFocused` is called inside `onFocusChange`, are you sure the `focused` argument is a boolean?

**Comment by meerim1987** [comment]
You can see that its initial value is `boolean`. But then it sets an object in `onFocusChange`. Otherwise if to pass boolean, like in my above example, the component stops focusing on selected month and day . I described everything above

**Comment by ljharb** [comment]
Right - what I mean is, it's the `onFocusChange` that's calling `setFocused` with a non-boolean argument.

Can you change it to:
```jsx
          onFocusChange={focused => {
            console.log(focused);
            setFocused(focused);
          }}
```
and see what gets logged?

**Comment by meerim1987** [comment]
I see the Object as expected. So when I distruct `focused` and pass the value to `setFocused` , we won't see the warning. But then the`DayPickerSingleDateController` component does not work as expected. Only when I pass this object it works well, month is focused with the day, but then we see the warning I attached above. 

<img width="351" alt="Screen Shot 2021-07-23 at 3 31 17 PM" src="https://user-images.githubusercontent.com/44931882/126847507-367740cc-e938-4bb5-acfa-9b5c9ee5c5f1.png">

**Comment by ljharb** [comment]
I see - so you're saying that the PropType wants a boolean, but the component only works properly when you pass `{ focused: <boolean> }`?

**Comment by ljharb** [comment]
So, the `onFocusChange` callback is *supposed* to get an object with a `focused` property - https://github.com/airbnb/react-dates/blob/master/src/components/DayPickerSingleDateController.jsx#L373 -which means your callback should look like this:
```jsx
          onFocusChange={({ focused }) => {
            setFocused(focused);
          }}
```

but if you're saying things don't work when you do that, then I'm not sure how to square that with our tests and with the way the component code is authored.

**Comment by meerim1987** [comment]
Yes, this was added ` onFocusChange={({ focused }) => {
            setFocused(focused);
          }}` as described in my issue above, it solved warning issue but broke the component. It was losing focus, much more like described in this thred, https://github.com/airbnb/react-dates/issues/709

**Comment by ljharb** [comment]
It seems like the real issue, then, is #709 - the propType error is correct.

**Comment by meerim1987** [comment]
I think this is another issue. It has similar consequances when component not working (when you pass a boolean). Though it should work correctly when you pass boolean. And in the other issue they are not talking about what triggers this component behave like that while here we are talking about this reason. So I think we can keep it as another issue/reason what makes it behave incorrectly.

**Comment by ljharb** [comment]
I agree it should work properly when passed a boolean, and that this issue is slightly different.
